include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -44,6 +54,97 @@ class __radix_sort_reorder_peer_kernel;
 template <::std::uint32_t, bool, bool, typename... _Name>
 class __radix_sort_reorder_kernel;
 
+//------------------------------------------------------------------------
+// radix sort: bitwise order-preserving conversions to unsigned integrals
+//------------------------------------------------------------------------
+
+#if __cpluslplus >= 202002L && __has_include(<bit>)
+template <typename _Dst, typename _Src>
+using __dpl_bit_cast = std::bit_cast<_Dst, _Src>;
+
+#elif __has_builtin(__builtin_bit_cast)
+template <typename _Dst, typename _Src>
+_Dst
+__dpl_bit_cast(const _Src& __src)
+{
+    static_assert(sizeof(_Dst) == sizeof(_Src), "Bit conversion for types of different size");
+    return __builtin_bit_cast(_Dst, __src);
+}
+
+#else
+template <typename _Dst, typename _Src>
+_Dst
+__dpl_bit_cast(const _Src& __src)
+{
+    static_assert(sizeof(_Dst) == sizeof(_Src), "Bit conversion for types of different size");
+    _Dst __result;
+    std::memcpy(&__result, &__src, sizeof(_Dst));
+    return __result;
+}
+#endif
+
+template <bool __is_ascending>
+bool
+__order_preserving_cast(bool __val)
+{
+    if constexpr (__is_ascending)
+        return __val;
+    else
+        return !__val;
+}
+
+template <bool __is_ascending, typename _UInt, typename = __enable_if_t<::std::is_unsigned_v<_UInt>>>
```

Comment:
Does it make sense to rethink `enable_if` approach we use? I know you don't like to use it as a return type but there is a compromised solution with dummy parameter but more "safe". This code compiles because you have different return types, otherwise it would compile-time error. Sure, today it works but I would suggest to use a little bit different approach for the future if somebody would edit this code and would not struggle with unobvious compilation error.

```suggestion
__enable_if_t<::std::is_unsigned_v<_UInt>, int> = 0>
```